overlord/boot/firstboot.go

```diff
@@ -29,34 +28,22 @@ import (
 	"github.com/snapcore/snapd/asserts"
 	"github.com/snapcore/snapd/asserts/snapasserts"
 	"github.com/snapcore/snapd/dirs"
-	"github.com/snapcore/snapd/firstboot"
-	"github.com/snapcore/snapd/logger"
-	"github.com/snapcore/snapd/osutil"
-	"github.com/snapcore/snapd/overlord"
 	"github.com/snapcore/snapd/overlord/assertstate"
 	"github.com/snapcore/snapd/overlord/auth"
 	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/overlord/state"
-	"github.com/snapcore/snapd/release"
 	"github.com/snapcore/snapd/snap"
 )
 
-var (
-	// ErrNotFirstBoot is an error that indicates that the first boot has already
-	// run
-	ErrNotFirstBoot = errors.New("this is not your first boot")
-)
-
-func populateStateFromSeed() error {
-	if osutil.FileExists(dirs.SnapStateFile) {
-		return fmt.Errorf("cannot create state: state %q already exists", dirs.SnapStateFile)
-	}
-
-	ovld, err := overlord.New()
+func PopulateStateFromSeed(st *state.State) error {
+	// check that the state is empty
+	all, err := snapstate.All(st)
```

Comment:
Should we check something more explicit, such as as "seeded" flag?  I'm concerned about the possibility of a seeded state that has no snaps (think classic).